packages/p2p/src/components/modal-manager/modals/quick-add-modal/quick-add-modal-content/add-payment-methods/add-payment-methods.tsx

```diff
@@ -0,0 +1,54 @@
+import React from 'react';
+import { Text } from '@deriv/components';
+import { isMobile } from '@deriv/shared';
+import { observer } from '@deriv/stores';
+import { useStores } from 'Stores';
+import { Localize } from 'Components/i18next';
+import AddPaymentMethod from 'Pages/my-profile/payment-methods/add-payment-method/add-payment-method.jsx';
+import SellAdPaymentMethodsList from 'Pages/my-ads/sell-ad-payment-methods-list.jsx';
+
+type TAddPaymentMethodsProps = {
+    selected_methods: string[];
+    setSelectedMethods: React.Dispatch<React.SetStateAction<string[]>>;
+};
+
+const AddPaymentMethods = ({ selected_methods, setSelectedMethods }: TAddPaymentMethodsProps) => {
+    const { my_ads_store } = useStores();
+
+    const { payment_method_ids, should_show_add_payment_method, setShouldShowAddPaymentMethod } = my_ads_store;
+
+    const onClickPaymentMethodCard = (payment_method: { ID: string }) => {
+        if (!payment_method_ids.includes(payment_method.ID)) {
+            if (payment_method_ids.length < 3) {
+                my_ads_store.payment_method_ids.push(payment_method.ID);
```

Comment:
shouldn't call this as just `payment_method_ids` as there were issues I had done it that way. This is because when you destructure a MobX observable, you are creating a new reference to the value of the observable. This new reference is not an observable itself, so it doesn't have the same reactivity as the original observable. Sooooo best to leave any observables that are being mutated alone. 

`This is also repeating in other components if you're wondering why its not destructured out`